bundles/umljava/tools.vitruv.applications.umljava.uml2java/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions

```diff
@@ -447,7 +448,30 @@ routine addUmlModelCorrespondence(uml::Model newModel){
 
 reaction UmlPackageCreated {
     after element uml::Package created and inserted in uml::Package[packagedElement]
-    call createJavaPackage(newValue, affectedEObject)
+    call createOrFindJavaPackage(newValue, affectedEObject)
+}
+
+routine createOrFindJavaPackage(uml::Package uPackage, uml::Package uSuperPackage) {
+	match {
+		require absence of java::Package corresponding to uPackage
+		val allPackages = retrieve many java::Package corresponding to ContainersPackage.Literals.PACKAGE
+	}
+	action {
+		call {
+			val matchingPackages = allPackages.filter[namespacesAsString + name == getUmlNamespaceAsString(uPackage)]
```

Comment:
See topmost comment: It should be possible to define this in a `with` block of the retrieval.